scrapy_zyte_api/providers.py

```diff
@@ -119,7 +119,12 @@ async def __call__(  # noqa: C901
                         "Actions dependencies must be annotated, "
                         "e.g. Annotated[Actions, actions([...list of actions...])]."
                     )
-                zyte_api_meta["actions"] = [dict(action) for action in cls.__metadata__[0]]  # type: ignore[attr-defined]
+                zyte_api_meta["actions"] = [
+                    {
+                        k: dict(v) if isinstance(v, frozenset) else v
```

Comment:
Oh, I think I didn't notice the added code is about the reverse conversion :)

I think this doesn't change tuples back to lists? I would expect dicts and lists in the result as it's a wrapper for a JSON API.

Reply:
The final result here would be a List of dicts. Do we need something different?

Reply:
Dict values can be lists, e.g. in the `values` key (looking at `Action` fields, which should match the API).

Reply:
Added a check for tuples on https://github.com/scrapy-plugins/scrapy-zyte-api/pull/188/commits/213182dd77ccfd7df4126040e3979d575db82df6